### PR TITLE
[#91211506] Proposals Controller

### PR DESCRIPTION
**Pull request by cmc333333**
The bulk of this PR is switching all of the `/carts` urls to be `/proposals`. Associated changes:

* Carts urls still exist, redirecting the user to the appropriate proposal url
* Replaced `cart_template_name` and `prefix_template_name` with `client_partial` equivalents
* Removed `CartsHelper` as its presentation logic was only used in one template and is not terribly complicated
* Cart decorator is now a Proposal decorator
* Compiled a few, single-use partials inline

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -1,32 +1,16 @@
-# @todo Rename
+# @todo Eventually, delete
```
Comment (afeld) [inline app/controllers/carts_controller.rb]: I think we can delete this controller now and do the redirection in `routes.rb`:

http://guides.rubyonrails.org/routing.html#redirection

File: app/views/proposals/_proposal_list.html.erb [hunk 2]
```diff
@@ -17,11 +17,20 @@
...
-      <td class="sixth"><a href="/carts/<%= proposal.cart.id %>"><%= proposal.cart.id %></a></td>
-      <td class="first"><a href="/carts/<%= proposal.cart.id %>"><%= proposal.name %></a></td>
+      <td class="sixth"><a href="<%= proposal_url(proposal) %>"><%= proposal.public_identifier %></a></td>
+      <td class="first"><a href="<%= proposal_url(proposal) %>"><%= proposal.name %></a></td>
       <td class="sixth"><%= number_to_currency(proposal.total_price) %></td>
       <td class="sixth">
-        <p><%= display_status(proposal.cart, current_user) %></p>
+        <% approvers = proposal.currently_awaiting_approvers %>
+        <% if proposal.pending? && approvers.include?(current_user) %>
+          <%# This isn't used anywhere %>
```
Comment (afeld) [inline app/views/proposals/_proposal_list.html.erb]: What do you mean? That condition would be `true` for an outstanding approval, right?

File: app/views/proposals/show.html.erb [hunk 3]
```diff
@@ -0,0 +1 @@
+<%= render partial: "proposal", locals: { proposal: @proposal}  %>
```
Comment (afeld) [inline app/views/proposals/show.html.erb]: Hmm, is that partial used in multiple places? Maybe we can just move that file in here.

**Comment by afeld** [comment]
This is great, thanks for doing it!

File: app/views/proposals/_proposal_list.html.erb (hunk 2 above)
Comment (cmc333333) [inline app/views/proposals/_proposal_list.html.erb]: The condition would be true if the user were an approver on an outstanding approval, but we only list carts by the requester. As the requester shouldn't be an approver on his/her own cart, I don't think this text is ever shown.

File: app/controllers/carts_controller.rb (hunk 1 above)
Comment (cmc333333) [inline app/controllers/carts_controller.rb]: Nice; can do.

File: app/views/proposals/_proposal_list.html.erb (hunk 2 above)
Comment (afeld) [inline app/views/proposals/_proposal_list.html.erb]: > we only list carts by the requester

Aaaah, that didn't used to be the case...I missed that change. [Storied](https://www.pivotaltracker.com/story/show/92792448).

File: app/views/proposals/show.html.erb (hunk 3 above)
Comment (cmc333333) [inline app/views/proposals/show.html.erb]: Good call

**Comment by cmc333333** [comment]
Updated to remove the cart controller and embed the partial, as it was only used in one template.

**Comment by afeld** [comment]
Needs merge from `master`, but otherwise :shipit:!

**Comment by cmc333333** [comment]
Merged master

**Pull request merged**